src/Language/PureScript/DCE/CoreFn.hs

```diff
@@ -111,24 +111,39 @@ runDeadCodeElimination entryPoints modules = uncurry runModuleDeadCodeEliminatio
 
       -- | Find dependencies of an expression.
       deps :: Expr Ann -> [Key]
-      deps = go
+      deps = onExpr'
         where
-          (_, go, _, _) = everythingOnValues (++)
-            (const [])
-            onExpr
-            onBinder
-            (const [])
-
           -- | Build graph from qualified identifiers.
           onExpr :: Expr Ann -> [Key]
           onExpr (Var _ i) = [i | isQualified i]
           onExpr _ = []
 
-          onBinder :: Binder Ann -> [Key]
-          onBinder (ConstructorBinder _ _ c _) =
-            [fmap (Ident . runProperName) c]
+          onExpr' :: Expr Ann -> [Key]
+          onExpr' v@(Literal _ l) = foldl (++) (onExpr v) (map onExpr' (extractLiteral l))
```

Comment:
What's the reason for the lazy `foldl`?  Similarly, for other left folds.

Reply:
It's a direct copy from the function removed in purescript/purescript#4148. I can make them `foldl'`

Reply:
ok, let's use strict fold.